### PR TITLE
Feature/data-landing

### PR DESCRIPTION
**Pull request by noahmanger**
Updates the data landing page template based on the new designs. Includes:
- New `GroupedBarChart` component 
- A mock data function to add the top-level totals from the server 

image
# Remaining to do:
- [x] Finish tooltips
- [x] Add pattern background for in-progress bars
- [x] Update the search area based on the new designs
- [x] Add the top-5 lists
- [ ] Add links to filings from the totals

Depends on https://github.com/18F/fec-style/pull/392

### CODE REVIEW

**Comment by noahmanger** [comment]
Update on this:

## What's done:
I have the hero, search area, links to advanced and election pages done:
image

The raising overview is done:
image

As is the spending overview:
image

## What's not done
- Need "about this data" content (new issue here: https://github.com/18F/openFEC-web-app/issues/1275)
- Need to complete the interaction of the feedback widget (https://github.com/18F/openFEC-web-app/issues/1249)
- The top candidate, pac and party committee lists are incomplete. I have the markup, interactivity and styles coded up, but we're lacking the necessary data, so I've removed it from this branch for now

Those three things can be dealt with separately after merging this in.

**Comment by adborden** [comment]
Has @jenniferthibault reviewed?

File: openfecwebapp/templates/landing.html [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  <section class="content__section--ruled content__section--extra">
+    <div class="grid grid--2-wide">
+      <div class="grid__item grid__item--no-margin card card--wide">
+        <div class="card__image__container">
+          <img class="card__image icon--complex" src="{{ url_for('static', filename='img/i-elections--neutral.svg') }}"  alt="Icon representing elections">
+          <h2 class="card__title"><a href="{{ url_for('election_lookup') }}">Elections by location &raquo;</a></h2>
+        </div>
+        <div class="card__content">
+          <h2 class="card__title"><a href="{{ url_for('election_lookup') }}">Elections by location &raquo;</a></h2>
+          <span>Find elections near you— search by state or ZIP code.</span>
```
Comment (adborden) [inline openfecwebapp/templates/landing.html]: I don't think there is supposed to be space after the dash. cc @emileighoutlaw ?

File: static/js/modules/bar-charts.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        '</pattern>' +
+      '</defs>' +
+  '</svg>';
+
+
+function GroupedBarChart(selector, data) {
+  this.element = d3.select(selector);
+  this.data = data;
+
+  this.margin = {top: 0, right: 20, bottom: 50, left: 40};
```
Comment (adborden) [inline static/js/modules/bar-charts.js]: Can these be pulled from the computed styles? e.g. using `jQuery.css`?

File: static/js/modules/bar-charts.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+GroupedBarChart.prototype.appendTooltip = function() {
+  return this.element
+    .style({'position': 'relative'})
+    .append('div')
+    .attr('class', 'tooltip tooltip--under tooltip--chart')
+    .attr('id', 'tooltip');
+};
+
+GroupedBarChart.prototype.populateTooltip = function(d) {
+  var entityMap = {
```
Comment (adborden) [inline static/js/modules/bar-charts.js]: Not a big deal, but when mapping internal names to UI names, I prefer to use the term "display names" so it's clear that these are meant for display to the user, not meant for other computation. I might consider naming this "entityDisplayNamesMap". I know, verbose :)

File: static/js/modules/bar-charts.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+GroupedBarChart.prototype.showTooltip = function(x0, d) {
+  var top = this.height + this.margin.top + this.margin.bottom;
+  var left = x0(d.period);
+  var content = this.populateTooltip(d);
+
+  this.tooltip
+    .style({'display': 'block'})
+    .style({'top': top.toString() + 'px'})
+    .style({'left': left.toString() + 'px'})
```
Comment (adborden) [inline static/js/modules/bar-charts.js]: This is a jquery element, right? Can you use the dictionary assignment:
```
.style({
  'display': 'block',
  'top': top.toString() + 'px',
  //...
})
```

File: static/js/modules/helpers.js [hunk 5]
```diff
@@ -180,6 +213,9 @@ module.exports = {
   filterNull: filterNull,
   buildAppUrl: buildAppUrl,
   buildUrl: buildUrl,
+  isMediumScreen: isMediumScreen,
+  isLargeScreen: isLargeScreen,
   globals: globals,
-  getTimePeriod: getTimePeriod
+  getTimePeriod: getTimePeriod,
+  zeroPad: zeroPad
```
Comment (adborden) [inline static/js/modules/helpers.js]: Not a big deal, maybe alphabetize this list so it's easier to spot duplicates and find what you're looking for?

**Comment by adborden** [comment]
👍

**Comment by jenniferthibault** [comment]
Haven't reviewed yet, plz don't merge. @noahmanger would you prefer to pair, or should I leave any styling change requests here?

**Comment by jenniferthibault** [comment]
Got a few things mostly in the spacing and font styles here, but also a few copy things from Emileigh's earlier feedback on the [visual design issue](https://github.com/18F/openFEC/issues/1656#issuecomment-225319559).

screen shot 2016-06-16 at 3 31 01 pm

The h3 styles aren't perfect headings here for "Raising overview" and "Spending overview", but I can address that later. 

Bigger fix: the `?` icon on the feedback thing isn't showing up
screen shot 2016-06-16 at 3 33 41 pm

**Comment by emileighoutlaw** [comment]
Oh nice! You make my words look so great, Jen!

File: static/js/modules/bar-charts.js (hunk 2 above)
Comment (noahmanger) [inline static/js/modules/bar-charts.js]: Mmm I don't think so. They're the _future_ properties of the `svg` element created below, and are used to calculate various things before that's appended.

**Comment by noahmanger** [comment]
Ok! Addressed the feedback. Thanks for the notes -- good stuff.

**Comment by noahmanger** [comment]
Update 2: The reaction box now works. If the user just clicks a reaction button, it sends a google analytics custom event. If they fill out the text box, it logs a feedback issue.

see

**Comment by jenniferthibault** [comment]
Noiiiiiice! 🎈 ⚡

File: static/js/modules/reaction-box.js [hunk 6]
```diff
@@ -3,36 +3,95 @@
...
-  ga('send', gaEventData);
+  if (typeof ga !== 'undefined') {
+    var gaEventData = {
+      hitType: 'event',
+      eventCategory: 'Reactions',
+      eventAction: this.location + '-' + this.name + ': ' + this.reaction,
+      eventValue: 1
+    };
+    ga('send', gaEventData);
+  }
```
Comment (adborden) [inline static/js/modules/reaction-box.js]: This should really be a helper method that would detect `ga` and do a no-op if it's unavailable (or even queue it if there's a possibility ga will load later).

File: static/js/modules/reaction-box.js [hunk 7]
```diff
@@ -3,36 +3,95 @@
...
 ReactionBox.prototype.showTextarea = function() {
-  this.$radios.attr('aria-hidden', true);
-  this.$textarea.attr('aria-hidden', false);
+  this.$step1.attr('aria-hidden', true);
+  this.$step2.attr('aria-hidden', false);
+
+  var labelMap = {
+    'informative': 'Great! \n What did you learn?',
+    'confusing': 'We\'re sorry to hear that. What didn\'t make sense?',
+    'not-interested': 'We\'re sorry to hear that. What would you like to see?',
```
Comment (adborden) [inline static/js/modules/reaction-box.js]: Should these be smart quotes?

File: static/js/modules/reaction-box.js [hunk 8]
```diff
@@ -3,36 +3,95 @@
...
+};
+
+ReactionBox.prototype.handleSubmit = function(e) {
+  e.preventDefault();
+  var data = {
+    action: 'Looking at the ' + '"' + this.name + '" chart on the ' +
+      this.location + ' page',
+    feedback: 'How was it? ' + this.reaction + '\n' +
+      'Tell us more: ' + this.$textarea.val(),
+  };
```
Comment (adborden) [inline static/js/modules/reaction-box.js]: I would prefer to do this kind of formatting on the server side. This allows the API to be more flexible if you're just specifying simple data attributes. e.g.
```javascript
{
  "chart_name": this.name,
  "location":  this.location,
  "reaction": this.reaction,
  "feedback": this.$textarea.val()
}
```

**Pull request merged**